packages/webapp/src/app/api/podcasts/query/route.tsx

```diff
@@ -1,31 +1,128 @@
+            with Markdown links to Wikipedia articles about people, places,
+            things, or events mentioned in the text. For example, if there is a
+            mention of China, you could replace the text with{' '}
+            {`“[China](https://en.wikipedia.org/wiki/China)“`}. Only return the
+            rewritten text, without any explanation. Do not include{' '}
+            {'"Passage:"'} in the response.
+          </SystemMessage>
+          <UserMessage>
+            Passage:{'\n'}
+            {/* <OpenAI chatModel="gpt-3.5-turbo-16k-0613"> */}
+            <OpenAI chatModel="gpt-4">
+              <ChatCompletion>
+                <SystemMessage>
+                  You are a helpful podcast chatbot and a trained question
+                  answerer. The user asks you a question about the podcast and
+                  you should answer the question truthfully, using only
+                  documents below that you retrieved from your internal memory.
+                  The documents are relevant pieces from the podcast transcript.
+                  Do not use any other knowledge you have about the world. If
+                  you {"don't"} know how to answer the question, just say
+                  {` "I don't know."`}.{'\n'}
+                  Note that the user cannot see the contents of the memory
+                  items. Also note that if an information is not present, it
+                  might be because your memory is incomplete.
+                  {'\n'}
+                  {/* If any sources are particularly relevant, make sure to cite
+            them by using a Markdown link format, at the end of your
+            response.{'\n'} */}
+                  {/* Whenever you reference a person, place, thing, or event, try to link
+            to the corresponding Wikipedia article, using a Markdown link. For
+            example, if there is a mention of China, you could replace the text
+            with {`“[China](https://en.wikipedia.org/wiki/China)“`}
+            {'\n'} */}
+                  If you find interesting follow up questions that the user can
+                  ask, you can include them at the very end of your response.
+                  {'\n'}
+                  Here are the relevant pieces of memory you have retrieved:
+                  {'\n```memory-pieces\n'}
+                  {chunks.map(chunk => chunkFormatter({ doc: chunk }))}
+                  {'\n```\n'}
+                  And here is the question you must answer:
+                </SystemMessage>
+                <UserMessage>{props.question}</UserMessage>
+              </ChatCompletion>
+            </OpenAI>
+          </UserMessage>
+        </ChatCompletion>
+      </OpenAI>
+      {'<table><tr>'}
+      {chunks.map(
+        async (doc, idx) =>
+          `\n<td key={${idx}}><details><summary>Doc #${idx}</summary>Source: ${doc.chunk.metadata?.source}<br/><br/>${doc.chunk.content}</details></td>`
+      )}
+      {'</tr></table>'}
```

Comment:
This is interesting – does this actually work, or are you just playing around with what UI would look like? :smile:

Reply:
It does something (see the carets at the bottom of the image):
![image](https://github.com/fixie-ai/podverse/assets/5062458/4ddd1b85-8637-4816-986b-b835ef82070a)

I'm kinda leaving the UI part to Matt, but at least the idea was that it gives some visibility into what is happening underneath.
This also allows us to maybe highlight the parts of the document that are relevant. Not sure how the UI would look like though. Just exposing it for now.